### PR TITLE
[ENH] [DOC] Add matlab to python file conversion functions, add docs for custom tractography integration

### LINKED ISSUE #549: Add Recommended Usage for Converting from MATLAB AFQ to pyAFQ
This could include:

- usage of [pyAFQ-scripts](https://github.com/36000/pyAFQ-scripts)

- recommended processing:

> Do your preprocessing and also reconstruction with another software, and now you’d like to feed it into pyAFQ.

### PR DESCRIPTION
**Pull request by 36000**
Closes #549 

- [x] Add matlab to pyhton converters from pyAFQ-scripts
- [x] Add tests for converters
- [x] Add docs for m2p converter
- [x] add docs for custom tractography

### CODE REVIEW

**Comment by 36000** [comment]
@arokem is there a small matlab file I could use to unit test these new functions?

**Comment by 36000** [comment]
Try this: https://github.com/yeatmanlab/AFQ-Browser/blob/master/afqbrowser/site/client/data/afq.mat

**Comment by 36000** [comment]
@arokem where should I put the mat files I use for testing here?

**Comment by arokem** [comment]
You can put them in a `tests/data/` folder, but you'll have to make sure to provide metadata that will tell Python to install the data as part of the package. See: https://github.com/nipy/nibabel/blob/master/setup.cfg#L84

**Comment by 36000** [comment]
I believe thats being used already for some s3 tests. I may have to modify those tests to use that folder.

For example, it looks like test_get_matching_s3_keys expects everything in that folder to be the same as whats in the test bucket.

@richford anything you would like to add to this discussion about how AFQ/tests/data is used?

**Comment by 36000** [comment]
I ended up moving the mock s3 data into a "mocks3" folder in data to separate it from the rest.

Anyways, this is ready for review / merge @arokem .

File: docs/source/usage/converter.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+takes in a path to the mAFQ tractography file and a path to an image.
+The function returns a Dipy Stateful Tractogram using the image as reference.
+Here is an example::
+    from AFQ.utils.conversion import matlab_tractography
+    sft = matlab_tractography("WholeBrainFG-1106.mat", "sub-1106.dwi.nii.gz")
+
+To convert an mAFQ mori groups file, use :func:`AFQ.utils.conversion.matlab_mori_groups`. This function
+takes in a path to the mAFQ mori groups file and a path to an image.
+The function returns a dictionary where keys are the pyAFQ bundle names and
+values are Dipy Stateful Tractograms using the image as reference. This is
```
Comment (arokem) [inline docs/source/usage/converter.rst:16]:
```suggestion
values are DIPY :class:`StatefulTractogram` class instances using the image as reference. This is
```

File: docs/source/usage/converter.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+the same structure of the output of pyAFQ's segmentation. If one wants to
+convert this dictionary to a single Stateful Tractogram, use bundles_to_tgram.
```
Comment (arokem) [inline docs/source/usage/converter.rst:18]:
```suggestion
convert this dictionary to a single :class:`StatefulTractogram` object, use :func:`bundles_to_tgram`.
```

File: docs/source/usage/converter.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Here is an example::
+    from AFQ.utils.conversion import matlab_mori_groups
+    fg = matlab_mori_groups("MoriGroups-1106.mat", "sub-1106.dwi.nii.gz")
```
Comment (arokem) [inline docs/source/usage/converter.rst:21]: But this example doesn't use `bundles_to_tgram`

File: docs/source/usage/converter.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pyAFQ can use tractography from other pipelines. To tell pyAFQ to use
+tractography from another pipeline, use the custom_tractography_bids_filters
+argument in the AFQ object or in the configuration file. This argument expects
+a dictionary of BIDS filters. pyAFQ will use these BIDS filters to find the
+tractography for each subject in each session.
+Here is an example custom_tractography_bids_filters::
+    custom_tractography_bids_filters = {'scope': 'qsiprep', 'suffix': 'tractography'}
+This would look for a file with the suffix 'tractography' inside of the
+'qsiprep' derivatives folder.
```
Comment (arokem) [inline docs/source/usage/converter.rst:36]: Maybe link to the now-integrated example?

File: AFQ/utils/conversion.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    ----------
+    mat_file : str
+        Path to a matlab tractography file.
+    img : Nifti1Image or str
+        Path to an img file to be loaded with nibabel or an img
+        to serve as the reference for the stateful tractogram.
+
+    Returns
+    -------
+    Dipy StatefulTractogram in RASMM space.
```
Comment (arokem) [inline AFQ/utils/conversion.py:43]:
```suggestion
    DIPY :class:`StatefulTractogram` in RASMM space.
```

File: AFQ/utils/conversion.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    """
+    mat_file = scipy.io.loadmat(mat_file)
```
Comment (arokem) [inline AFQ/utils/conversion.py:45]: I think that if you call `loadmat` with `squeeze_me=True` you con't have to index into singleton dimensions below.

File: AFQ/utils/conversion.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    img : Nifti1Image or str
+        Path to an img file to be loaded with nibabel or an img
+        to serve as the reference for the stateful tractogram.
+
+    Returns
+    -------
+    Dictionary where keys are the pyAFQ bundle names and values are
+    Dipy StatefulTractograms in RASMM space.
+    """
+    mat_file = scipy.io.loadmat(mat_file)
```
Comment (arokem) [inline AFQ/utils/conversion.py:75]: Same comment here.

File: AFQ/utils/conversion.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    mat_file : str
+        Path to a matlab mori groups file.
+    img : Nifti1Image or str
+        Path to an img file to be loaded with nibabel or an img
+        to serve as the reference for the stateful tractogram.
+
+    Returns
+    -------
+    Dictionary where keys are the pyAFQ bundle names and values are
+    Dipy StatefulTractograms in RASMM space.
```
Comment (arokem) [inline AFQ/utils/conversion.py:73]:
```suggestion
    DIPY :class:`StatefulTractogram` instances in RASMM space.
```

File: AFQ/utils/conversion.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    This dictionary is structured the same way as the results of pyAFQ
+    segmentation. The keys are bundle names and the values are stateful
+    tractograms.
+    If you want to merge this dictionary into one stateful tractogram,
+    use bundles_to_tgram.
+
+    Parameters
+    ----------
+    mat_file : str
+        Path to a matlab mori groups file.
```
Comment (arokem) [inline AFQ/utils/conversion.py:65]:
```suggestion
        Path to a matlab Mori groups file.
```

File: AFQ/utils/conversion.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    if isinstance(img, str):
+        img = nib.load(img)
+
+    tracker = MatlabFileTracking(mat_file['fg']['fibers'][0][0])
+    return StatefulTractogram(tracker, img, Space.RASMM)
+
+
+def matlab_mori_groups(mat_file, img):
+    """
+    Converts a matlab mori groups file to a dictionary of fiber groups.
```
Comment (arokem) [inline AFQ/utils/conversion.py:55]:
```suggestion
    Converts a matlab Mori groups file to a dictionary of fiber groups.
```

File: AFQ/utils/conversion.py [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    This dictionary is structured the same way as the results of pyAFQ
+    segmentation. The keys are bundle names and the values are stateful
+    tractograms.
```
Comment (arokem) [inline AFQ/utils/conversion.py:58]:
```suggestion
    segmentation. The keys are bundle names and the values are 
    :class:`StatefulTractogram` instances.
```

File: AFQ/utils/conversion.py [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    If you want to merge this dictionary into one stateful tractogram,
```
Comment (arokem) [inline AFQ/utils/conversion.py:59]:
```suggestion
    If you want to merge this dictionary into one :class:`StatefulTractogram`,
```

File: AFQ/utils/conversion.py [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+    use bundles_to_tgram.
```
Comment (arokem) [inline AFQ/utils/conversion.py:60]:
```suggestion
    use :func:`bundles_to_tgram`.
```

File: docs/source/usage/converter.rst [hunk 14]
```diff
@@ -0,0 +1,36 @@
+Matlab AFQ to Python AFQ conversion
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pyAFQ provides tools to convert between the results of the original Matlab
+based AFQ (which we call mAFQ) and pyAFQ at various stages in the pipeline.
+
+To convert an mAFQ tractography file, use :func:`AFQ.utils.conversion.matlab_tractography`. This function
+takes in a path to the mAFQ tractography file and a path to an image.
+The function returns a Dipy Stateful Tractogram using the image as reference.
```
Comment (arokem) [inline docs/source/usage/converter.rst:8]:
```suggestion
The function returns a DIPY :class:`StatefulTractogram` class instance using the image as reference.
```

File: docs/source/usage/converter.rst [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+Here is an example::
+    from AFQ.utils.conversion import matlab_tractography
+    sft = matlab_tractography("WholeBrainFG-1106.mat", "sub-1106.dwi.nii.gz")
+
+To convert an mAFQ mori groups file, use :func:`AFQ.utils.conversion.matlab_mori_groups`. This function
+takes in a path to the mAFQ mori groups file and a path to an image.
```
Comment (arokem) [inline docs/source/usage/converter.rst:14]:
```suggestion
takes in a path to the mAFQ Mori groups file and a path to an image.
```

**Review by arokem — COMMENTED** [review]
Overall, looks good! Just a few minor comments/suggestions.

**Comment by 36000** [comment]
@arokem I tried using squeeze_me but I got confused by what it was returning. The current system works for the files I have encountered, so I don't know if it would be useful to re-figure it out with squeeze_me.

Other than that, this is ready for review / merge.

**Pull request merged**